packages/babel-plugin-transform-pipeline-operator/test/fixtures/pipeline-operator/arrow-functions/expected.js

```diff
@@ -0,0 +1,15 @@
+var _ref, _ref2, _sum;
+
+var result = (_ref = [5, 10], (_ref2 = _ref.map(x => x * 2), (_sum = _ref2.reduce((a, b) => a + b), _sum + 1)));
+assert.equal(result, 31);
+
+var inc = x => x + 1;
+
+var double = x => x * 2;
+
+var result2 = [4, 9].map(x => {
+  var _ref3, _x;
+
+  return _ref3 = (_x = x, inc(_x)), double(_ref3);
```

Comment:
This `_x` feels wasteful but I guess it works to guard against mutations on the `right` side (e.g. `++` operator)

Reply:
Yah, any getter that occurs during the `inc` evaluation (not the actual call, but the reference) could cause mutations to `x`.